app/src/full/java/io/homeassistant/companion/android/sensors/LocationSensorManager.kt

```diff
@@ -208,19 +223,31 @@ class LocationSensorManager : BroadcastReceiver(), SensorManager {
             if (Build.VERSION.SDK_INT >= 26) location.verticalAccuracyMeters.toInt() else 0
         )
 
+        val sensorDao = AppDatabase.getInstance(latestContext).sensorDao()
+        val locationEntity = sensorDao.get(backgroundLocation.id)
+
+        if (locationEntity?.state == updateLocation.gps.contentToString()) {
+            Log.d(TAG, "Same location as last update, not sending to HA")
```

Comment:
Not sending the location if it is the same as previous is a bad idea. If a person stays at one place the location most probably will be the same at all times. It makes impossible to distinguish if a person really stays at one place or his phone or home assistant app has died.

Reply:
This checks for an `exact` GPS match, even when sitting still the value will change slightly and be sent.

Reply:
> 
> 
> This checks for an `exact` GPS match, even when sitting still the value will change slightly and be sent.

Not true. Here is my log for example:

```
2020-09-01 15:45:02 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Andrey for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:47:03 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Andrey for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:49:03 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Andrey for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:49:59 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Andrey for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:50:28 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Andrey for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:50:28 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:52:34 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:54:35 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:54:59 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:56:35 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:58:36 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:59:59 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:00:36 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:02:37 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:04:38 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:04:59 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:05:29 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:05:29 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:07:39 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:09:40 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:09:59 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:11:40 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:13:41 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:14:59 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:15:59 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:18:07 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:19:59 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:20:07 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:21:02 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:21:02 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:23:08 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
```

Reply:
> 
> 
> This checks for an `exact` GPS match, even when sitting still the value will change slightly and be sent.

Not true. GPS works only if you a GPS navigation app open in the foreground. All other times cell-tower/wi-fi based geolocation is used. And the coordinates will be exactly the same at all times:
```
2020-09-01 15:45:02 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:47:03 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:49:03 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:49:59 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:50:28 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:50:28 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:52:34 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:54:35 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:54:59 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:56:35 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:58:36 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 15:59:59 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:00:36 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:02:37 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:04:38 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:04:59 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:05:29 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:05:29 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:07:39 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:09:40 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:09:59 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:11:40 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:13:41 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:14:59 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:15:59 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:18:07 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:19:59 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:20:07 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:21:02 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:21:02 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
2020-09-01 16:23:08 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Anna for type update_location: {'location_name': '', 'gps': [55.6373754, 37.3570549], 'gps_accuracy': 11, 'speed': 0, 'altitude': 201, 'course': 0, 'vertical_accuracy': 2}
```

Reply:
We could still send them, but throttle them to max once per 5 mins?

Reply:
Maybe we can filter out duplicates within 15 minutes of last location update being sent. That way it will be updating once people start moving, and if they stay home we still send every 15 min.

```
if (Now() - lastUpdate < 15min && isDuplicate) { return }
```

Reply:
Just pushed that exact logic. ;-p

Reply:
You already limited requestSingleAccurateLocation to once per minute, what is the purpose of adding additional limits?